lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -225,6 +225,9 @@ void swift::eraseUsesOfInstruction(SILInstruction *Inst,
   for (auto UI = Inst->use_begin(), E = Inst->use_end(); UI != E;) {
     auto *User = UI->getUser();
     UI++;
+    if (!User) {
```

Comment:
This looks like an unrelated fix, maybe it should be its own commit or PR?

Reply:
At least add a comment here to explain what is going on with the iterator. If this is correct, we are relying on some sort of special semantics to make sure we are memory safe.